### ISSUE TITLE
Curious about classification results without pre-training

### ISSUE THREAD

**Issue by LoveMiki** [Sha-Lab/FEAT#60]
Thanks for the excellent work. 

In your paper, I noticed that you pre-trained the backbone to initialize its weights rather than training the backbone from scratch, is that right?
As can be seen from Table 1, the pre-trained ProtoNet has achieved comparable performance to those of SOTA approaches, which makes me believe that the pre-training strategy has a big impact on the performance improvement. If so, will the classification accuracies drop without the pre-training? Are there any evaluation results of the proposed method without using the pre-training strategy from the beginning?

**Comment by Han-Jia** [comment]
Hi,

We find it is a bit hard to train the model (e.g., ProtoNet) from scratch. Usually, it requires a larger way during meta-training (e.g., using 1-shot 30-way tasks) or longer epochs. 

We find there is a simple and efficient approach that yields competitive results (around 62% for 1-shot 5-way tasks on MiniIMageNet for ProtoNet) when training from scratch. After we make a forward step of a mini-batch, we can **resplit** the tasks (the partition of support and query sets) multiple times, so that a single mini-batch can generate a lot of tasks. Optimizing over those tasks jointly (i.e., the averaged loss) can get a more stable gradient in the back-propagation. The notion is named as multi-classifier training for generalized few-shot learning in [this paper](https://arxiv.org/pdf/1906.02944.pdf) and sufficient episodic training for unsupervised meta-learning at [here](https://arxiv.org/pdf/2011.14663.pdf).

**Comment by LoveMiki** [comment]
> Hi,
> 
> We find it is a bit hard to train the model (e.g., ProtoNet) from scratch. Usually, it requires a larger way during meta-training (e.g., using 1-shot 30-way tasks) or longer epochs.
> 
> We find there is a simple and efficient approach that yields competitive results (around 62% for 1-shot 5-way tasks on MiniIMageNet for ProtoNet) when training from scratch. After we make a forward step of a mini-batch, we can **resplit** the tasks (the partition of support and query sets) multiple times, so that a single mini-batch can generate a lot of tasks. Optimizing over those tasks jointly (i.e., the averaged loss) can get a more stable gradient in the back-propagation. The notion is named as multi-classifier training for generalized few-shot learning in [this paper](https://arxiv.org/pdf/1906.02944.pdf) and sufficient episodic training for unsupervised meta-learning at [here](https://arxiv.org/pdf/2011.14663.pdf).

Thanks for the reply. But what i am concerned about is that the performance gap between FEAT and SimpleShot (with WRN backbone) on MiniImageNet dataset is so tiny (81.11 vs 80.33). This result could be attributed to the pre-training stragery of FEAT, but not the mechanism of set-to-set function. Also, the classification result of TieredImageNet under the ResNet backnone have the same phenomenon. Although FEAT used ResNet-12 whereas SimpleShot used ResNet-18 on TieredImageNet, the difference in embedding capability could be made up by the pre-training stragery. From these perspective, the results are not fully convincing to me (correct me if i am wrong).

Thank you.

**Comment by Han-Jia** [comment]
Hi,

Pre-training the backbone is important for few-shot learning, which was further empirically validated by some papers. 

Based on the same pre-trained weights on MiniImageNet w/ ResNet-12, SimpleShot can get around 65% while FEAT can get more than 66%. So meta-learning with FEAT is necessary to get improvements based on the pre-trained embedding.

**Comment by LoveMiki** [comment]
As mentioned previously, table 2 shown in your paper that the performance of FEAT (pre-train + transformer) only slightly higher than SimpleShot (without pre-train). That's what i am concerned about. And based on what you say here, I beileve that pre-training contributes much more on classification improvement than transformer.
